api/v1alpha1/envoypatchpolicy_types.go

```diff
@@ -0,0 +1,124 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+const (
+	// KindEnvoyPatchPolicy is the name of the EnvoyPatchPolicy kind.
+	KindEnvoyPatchPolicy = "EnvoyPatchPolicy"
+)
+
+// +kubebuilder:object:root=true
+
+// EnvoyPatchPolicy allows the user to modify the generated Envoy xDS
+// resources by Envoy Gateway using this patch API
+type EnvoyPatchPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of EnvoyPatchPolicy.
+	Spec EnvoyPatchPolicySpec `json:"spec"`
+
+	// Status defines the current status of EnvoyPatchPolicy.
+	Status EnvoyPatchPolicyStatus `json:"status"`
+}
+
+// EnvoyPatchPolicySpec defines the desired state of RateLimitFilter.
+// +union
+type EnvoyPatchPolicySpec struct {
+	// Type decides the type of patch.
+	// Valid EnvoyPatchType values are "JSONPatch".
+	//
+	// +unionDiscriminator
+	Type EnvoyPatchType `json:"type"`
+	// JSONPatch defines the JSONPatch configuration.
+	//
+	// +optional
+	JSONPatches []EnvoyJSONPatchConfig `json:"jsonPatches,omitempty"`
+	// TargetRef is the name of the Gateway API resource this policy
+	// is being attached to.
+	// Currently only attaching to Gateway is supported
+	// This Policy and the TargetRef MUST be in the same namespace
+	// for this Policy to have effect and be applied to the Gateway
+	// TargetRef
+	TargetRef gwapiv1a2.PolicyTargetReference `json:"targetRef"`
```

Comment:
Can I have a "all" target? There might be scenarios multiple gateway resources need to be apply the same EnvoyFilters.

Reply:
prefer if we started off with 1:1 mappings, since policy attachment UX is already perceived as complex, the use case can be satisfied today by creating another identical PolicyAttachment and applying it to another `TargetRef`. In the future we could make `TargetRef` plural, note this definition is taken from upstream